src/SwipeRating.tsx

```diff
@@ -69,116 +69,116 @@ export type SwipeRatingProps = {
    *
    * Default is 'star'
    */
-  type?: string;
+  type: string;
 
   /**
    * Pass in a custom image source; use this along with type='custom' prop above
    */
-  ratingImage?: React.ReactNode;
+  ratingImage: React.ReactNode;
```

Comment:
are we making all of these params as mandatory params?

Reply:
I have used Partial. Ref: https://www.typescriptlang.org/docs/handbook/utility-types.html

Reply:
Currently, all the props were optional. Therefore it's a good practice to use Partial here.

Reply:
Although most props are optional: the attribute is used to distinguish between required and optional params.

If there is a default value for all props - and they should be made optional- we should update all params to be optional instead of using Partial.

Reply:
In that case, closing the PR will work, right? @sshivananda